src/algebra/group_power/lemmas.lean

```diff
@@ -180,6 +181,138 @@ f.to_multiplicative.map_gpow a n
 
 end group
 
+namespace with_top
+
+section nsmul
+
+section add_monoid
+
+variables [add_monoid A] (n : ℕ) (x : with_top A)
+
+lemma nsmul_coe (x : A) :
+  n • (x : with_top A) = ((n • x : A) : with_top A) :=
+(add_monoid_hom.map_nsmul coe_add_hom _ _).symm
+
+@[simp] lemma succ_nsmul_eq_top_iff (n : ℕ) {x : with_top A} :
+  (n + 1) • x = ⊤ ↔ x = ⊤ :=
+begin
+  induction x using with_top.rec_top_coe,
+  { rw [succ_nsmul', with_top.add_top] },
+  { simp only [with_top.nsmul_coe, with_top.coe_ne_top] }
+end
+
+lemma smul_top_of_ne_zero (hx : 0 < n) :
+  n • (⊤ : with_top A) = ⊤ :=
+begin
+  obtain ⟨k, rfl⟩ := nat.exists_eq_succ_of_ne_zero hx.ne',
+  simp
+end
+
+@[simp] lemma succ_nsmul_top (n : ℕ) :
+  (n + 1) • (⊤ : with_top A) = ⊤ :=
+by simp
+
+end add_monoid
+
+lemma nsmul_neg [add_group A] (n : ℕ) (x : with_top A) :
+  n • - x = - (n • x) :=
+begin
+  induction x using with_top.rec_top_coe,
+  { cases n,
+    { simp [zero_nsmul] },
+    { simp } },
+  { rw [with_top.neg_coe, with_top.nsmul_coe, neg_nsmul, with_top.nsmul_coe,
+        with_top.neg_coe] }
+end
+
+end nsmul
+
+section gsmul
+
+section
+
+variable [add_group A]
+
+lemma gsmul_coe (r : ℤ) (x : A) :
+  r • (x : with_top A) = ((r • x : A) : with_top A) :=
+begin
+  induction r using int.induction_on with r IH r IH,
+  { simp },
+  { simp [←int.coe_nat_succ, nsmul_coe] },
+  { simp [←int.neg_succ_of_nat_coe', ←with_top.neg_coe, nsmul_coe] }
+end
+
+@[simp] lemma gsmul_zero (r : ℤ) :
+  r • (0 : with_top A) = 0 :=
+by rw [←with_top.coe_zero, with_top.gsmul_coe, gsmul_zero]
+
+lemma zero_gsmul (x : with_top A) :
+  (0 : ℤ) • x = 0 :=
+by simp
```

Comment:
Instead of having these two lemmas, I'd recommend just making a `smul_with_zero` instance